loopd/macaroons.go

```diff
@@ -0,0 +1,177 @@
+package loopd
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+const (
+	// loopMacaroonLocation is the value we use for the loopd macaroons'
+	// "Location" field when baking them.
+	loopMacaroonLocation = "loop"
+)
+
+var (
+	// RequiredPermissions is a map of all loop RPC methods and their
+	// required macaroon permissions to access loopd.
+	RequiredPermissions = map[string][]bakery.Op{
+		"/looprpc.SwapClient/LoopOut": {{
+			Entity: "swap",
+			Action: "execute",
+		}, {
+			Entity: "loop",
+			Action: "out",
+		}},
+		"/looprpc.SwapClient/LoopIn": {{
+			Entity: "swap",
+			Action: "execute",
+		}, {
+			Entity: "loop",
+			Action: "in",
+		}},
+		"/looprpc.SwapClient/Monitor": {{
+			Entity: "swap",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/ListSwaps": {{
+			Entity: "swap",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/SwapInfo": {{
+			Entity: "swap",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/LoopOutTerms": {{
+			Entity: "terms",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/LoopOutQuote": {{
+			Entity: "swap",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/GetLoopInTerms": {{
+			Entity: "terms",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/GetLoopInQuote": {{
+			Entity: "swap",
+			Action: "read",
+		}},
+		"/looprpc.SwapClient/GetLsatTokens": {{
+			Entity: "auth",
+			Action: "read",
+		}},
+	}
+
+	// allPermissions is the list of all existing permissions that exist
+	// for loopd's RPC. The default macaroon that is created on startup
+	// contains all these permissions and is therefore equivalent to lnd's
+	// admin.macaroon but for loop.
+	allPermissions = []bakery.Op{{
+		Entity: "loop",
+		Action: "out",
+	}, {
+		Entity: "loop",
+		Action: "in",
+	}, {
+		Entity: "swap",
+		Action: "execute",
+	}, {
+		Entity: "swap",
+		Action: "read",
+	}, {
+		Entity: "terms",
+		Action: "read",
+	}, {
+		Entity: "auth",
+		Action: "read",
+	}}
+
+	// macDbDefaultPw is the default encryption password used to encrypt the
+	// loop macaroon database.
+	//
+	// TODO(guggero): Allow the password to be specified by the user. Needs
+	// create/unlock calls in the RPC. Using a password should be optional
+	// though.
```

Comment:
Does something still need to be done with this? I'd think that a default password in open source code provides a false sense of security?

Reply:
I changed the password to be an empty one and added some more information to the comment.